backend/infrahub/git/integrator.py

```diff
@@ -126,6 +129,33 @@ class InfrahubRepositoryIntegrator(InfrahubRepositoryBase):  # pylint: disable=t
     class that uses an "InfrahubRepository" or "InfrahubReadOnlyRepository" as input
     """
 
+    @classmethod
+    async def init(cls, commit: str | None = None, service: InfrahubServices | None = None, **kwargs: Any) -> Self:
+        service = service or InfrahubServices()
+        self = cls(service=service, **kwargs)
+        try:
+            self.validate_local_directories()
+        except RepositoryInvalidFileSystemError:
+            await self.ensure_location_is_defined()
+            await self.create_locally(infrahub_branch_name=self.infrahub_branch_name, update_commit_value=False)
```

Comment:
`self.infrahub_branch_name` is an optional string, which seems to be set in some instances but not everywhere. Looking at the .create_locally() method we can see that it's an optional parameter and only used when `update_commit_value` is `True`. I'd remove it as a parameter here to avoid confusion.